src/sql/base/browser/ui/table/plugins/headerFilter.plugin.ts

```diff
@@ -117,36 +123,31 @@ export class HeaderFilter<T extends Slick.SlickData> {
 			.remove();
 	}
 
-	private addMenuItem(menu: JQuery<HTMLElement>, columnDef: Slick.Column<T>, title: string, command: HeaderFilterCommands) {
-		const $item = jQuery('<div class="slick-header-menuitem">')
-			.data('command', command)
-			.data('column', columnDef)
-			.bind('click', async (e) => await this.handleMenuItemClick(e, command, columnDef))
-			.appendTo(menu);
-
-		const $icon = jQuery('<div class="slick-header-menuicon">')
-			.appendTo($item);
-
-		if (title === 'Sort Ascending') {
-			$icon.get(0).className += ' ascending';
-		} else if (title === 'Sort Descending') {
-			$icon.get(0).className += ' descending';
-		}
-
-		jQuery('<span class="slick-header-menucontent">')
-			.text(title)
-			.appendTo($item);
+	private createButtonMenuItem(menu: HTMLElement, columnDef: Slick.Column<T>, title: string, command: HeaderFilterCommands, iconClass: string): Button {
+		const buttonContainer = document.createElement('div');
```

Comment:
it seems like jquery used to grab the old elements with the classes but we create a new `div` every time now. Do we also remove it from the `DOM` every time when we hide the context menu? Otherwise it will be pretty bad with performance

Reply:
the $menu element will be cleared when the dialog is closed, all the child elements will be disposed too.